keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1535 @@
+//
+// - If the requested amount is less than Minimum, it is rounded up to the Minimum value.
+// - If the requested amount is between Minimum and Maximum, ChunkSize is set,
+//   and the amount does not align with the ChunkSize,
+//   it will be rounded up to the next value matching Minimum + (n * ChunkSize).
+// - If the requested or rounded amount exceeds Maximum (if set), the request does not satisfy the policy,
+//   and the device cannot be allocated.
+//
+// - If ChunkSize is not set, the requested amount is used as-is, provided it falls within the range.
+type CapacitySharingPolicyRange struct {
+    // Minimum specifies the minimum capacity allowed for a consumption request.
+    //
+    // Minimum must be less than or equal to the capacity value.
+    // Default must be more than or equal to the minimum.
+    //
+    // +required
+    Minimum resource.Quantity
+
+    // Maximum defines the upper limit for capacity that can be requested.
+    //
+    // Maximum must be less than or equal to the capacity value.
+    // Minimum and default must be less than or equal to the maximum.
+    //
+    // +optional
+    Maximum *resource.Quantity
+
+    // ChunkSize defines the step size between valid capacity amounts within the range.
+    //
+    // Maximum and default must be a multiple of the chunk size.
+    // Minimum + chunk size must be less than or equal to the capacity value.
+    //
+    // +optional
+    ChunkSize *resource.Quantity
+}
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+The `CapacityRequests` field is defined within each supported device request type, such as `DeviceSubRequest` and `ExactDeviceRequest`.
+
+```go
+// DeviceSubRequest is similar to ExactDeviceRequest, but doesn't expose the
+// AdminAccess field as that one is only supported when requesting a
+// specific device.
+type DeviceSubRequest struct {
+...
+   // CapacityRequests define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   CapacityRequests *CapacityRequirements
+}
+
+// ExactDeviceRequest is a request for one or more identical devices.
+type ExactDeviceRequest struct {
+...
+   // CapacityRequests define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   CapacityRequests *CapacityRequirements
+}
+
+
+// CapacityRequirements defines the capacity requirements for a specific device request.
+type CapacityRequirements struct {
+    // Minimum defines the minimum amount of each device capacity required for the request.
+    // Each minimum amount must be a standard Quantity.
+    //
+    // If the capacity has a sharing policy, this value is rounded up to the nearest valid amount
+    // according to that policy. The rounded value is used during scheduling to determine how much capacity to consume.
+    //
+    // If the quantity does not have a sharing policy, this value is used as an additional filtering
```

Comment:
The filtering aspect happens whether or not there's a sharing policy defined, right? If a request asks for 5 widgets, devices that don't have 5 widgets of capacity are excluded.